libs/sdk-bindings/src/breez_sdk.udl

```diff
@@ -193,10 +193,20 @@ dictionary CheckMessageResponse {
     boolean is_valid;
 };
 
-enum PaymentTypeFilter {
-    "Sent",
-    "Received",
-    "ClosedChannel",
+[Enum]
+interface JsonFieldValue {
+    IsNull();
+    Bool(boolean value);
+    Number(u64 value);
+    String(string value);
+};
+
+[Enum]
+interface PaymentTypeFilter {
+    Sent();
+    Received();
+    ClosedChannel();
+    Metadata(string key, JsonFieldValue value);
```

Comment:
Perhaps we can simplify this and use strings only instead of JsonFieldValue ?

Reply:
Also we should think about this interface a bit more as the sent/received/closed channels are used as union in the query and metadata is used as filter.

Reply:
Addressed both points in the following commit. Would've been nice to use a `HashMap<String, String>` instead of a `Vec<PaymentMetadata>` for the filters, but apparently flutter_rust_bridge does not seem to support them (https://github.com/fzyzcjy/flutter_rust_bridge/issues/329).